src/init/initHelpers.cpp

```diff
@@ -3,3 +3,11 @@
 bool Init::isSudo() {
     return getuid() != geteuid() || !geteuid();
 }
+
+void Init::gainRealTime() {
+    const int   minPrio = sched_get_priority_min(SCHED_RR);
+    sched_param param   = {.sched_priority = minPrio};
+
+    if (pthread_setschedparam(pthread_self(), SCHED_RR | SCHED_RESET_ON_FORK, &param))
```

Comment:
Can you wrap behind `#ifndef SCHED_RESET_ON_FORK`? At least on FreeBSD [only root](https://github.com/freebsd/freebsd-src/blob/releng/13.2/sys/kern/kern_thr.c#L203-L210) can set [scheduler policy](https://github.com/freebsd/freebsd-src/blob/releng/13.2/lib/libthr/thread/thr_kern.c#L94-L97), and this fails to build:
```c++
src/init/initHelpers.cpp:11:58: error: use of undeclared identifier 'SCHED_RESET_ON_FORK'
    if (pthread_setschedparam(pthread_self(), SCHED_RR | SCHED_RESET_ON_FORK, &param))
                                                         ^
```
```
$ Hyprland -c /dev/null
User-specified config location: /dev/null
Welcome to Hyprland!
Failed to change process scheduling strategy
[...]
```
```diff
diff --git a/src/init/initHelpers.cpp b/src/init/initHelpers.cpp
index be1c17dd..97fe7c86 100644
--- a/src/init/initHelpers.cpp
+++ b/src/init/initHelpers.cpp
@@ -5,9 +5,11 @@ bool Init::isSudo() {
 }
 
 void Init::gainRealTime() {
+#ifdef SCHED_RESET_ON_FORK
     const int   minPrio = sched_get_priority_min(SCHED_RR);
     sched_param param   = {.sched_priority = minPrio};
 
     if (pthread_setschedparam(pthread_self(), SCHED_RR | SCHED_RESET_ON_FORK, &param))
         Debug::log(WARN, "Failed to change process scheduling strategy");
+#endif
 }
\ No newline at end of file
```

Reply:
To be accurate, `CAP_SYS_NICE` can probably be emulated via [mac_priority(4)](https://man.freebsd.org/mac_priority/4) but it's not enabled by default and without disabling inheritance it's not going to be useful here.

Reply:
done